jdbc-driver/src/test/java/com/google/cloud/dataproc/jdbc/DataprocInfoTest.java

```diff
@@ -200,65 +244,82 @@ public void getHost_noClusterFoundByName() {
+                        HOST_1, PORT, AUTHORIZATION, TOKEN_VALUE);
         HiveJdbcConnectionOptions param = HiveUrlUtils.parseHiveUrl(simpleUrl);
-        DataprocInfo infoTest = new DataprocInfo(param, mockClusterControllerClient);
+        DataprocInfo infoTest = new DataprocInfo(param, mockClusterControllerClient, mockCredentials);
         assertThat(infoTest.toHiveJdbcUrl()).isEqualTo(hiveUrl);
     }
 
     @Test
     public void toHiveJdbcUrl_longClusterName() throws SQLException {
         String longUrlComplete =
-                "jdbc:dataproc://hive/db-name;port=10003;projectId=pid;region=us-central1;clusterName=simple-cluster2;user=foo;password=bar?hive.support.concurrency=true#a=123";
+                "jdbc:dataproc://hive/db-name;projectId=pid;region=us-central1;clusterName=simple-cluster2;user=foo;password=bar?hive.support.concurrency=true#a=123";
         String hiveUrl =
-                "jdbc:hive2://simple-cluster2-m:10003/db-name;transportMode=http;httpPath=cliservice;user=foo;password=bar?hive.support.concurrency=true#a=123";
+                String.format(
+                        "jdbc:hive2://%s:%d/db-name;transportMode=http;httpPath=cliservice;%s%s;user=foo;password=bar?hive.support.concurrency=true#a=123",
+                        HOST_2, PORT, AUTHORIZATION, TOKEN_VALUE);
         HiveJdbcConnectionOptions param = HiveUrlUtils.parseHiveUrl(longUrlComplete);
-        DataprocInfo infoTest = new DataprocInfo(param, mockClusterControllerClient);
+        DataprocInfo infoTest = new DataprocInfo(param, mockClusterControllerClient, mockCredentials);
         assertThat(infoTest.toHiveJdbcUrl()).isEqualTo(hiveUrl);
     }
 
     @Test
     public void toHiveJdbcUrl_longClusterLabel() throws SQLException {
         String longUrlComplete =
-                "jdbc:dataproc://hive/db-name;port=10003;projectId=pid;region=us-central1;clusterPoolLabel=com=google:env=staging:team=dataproc;user=foo;password=bar?hive.support.concurrency=true#a=123";
+                "jdbc:dataproc://hive/db-name;projectId=pid;region=us-central1;clusterPoolLabel=com=google:env=staging:team=dataproc;user=foo;password=bar?hive.support.concurrency=true#a=123";
         String hiveUrl =
-                "jdbc:hive2://simple-cluster1-m:10003/db-name;transportMode=http;httpPath=cliservice;user=foo;password=bar?hive.support.concurrency=true#a=123";
+                String.format(
+                        "jdbc:hive2://%s:%d/db-name;transportMode=http;httpPath=cliservice;%s%s;user=foo;password=bar?hive.support.concurrency=true#a=123",
+                        HOST_1, PORT, AUTHORIZATION, TOKEN_VALUE);
         HiveJdbcConnectionOptions param = HiveUrlUtils.parseHiveUrl(longUrlComplete);
-        DataprocInfo infoTest = new DataprocInfo(param, mockClusterControllerClient);
+        DataprocInfo infoTest = new DataprocInfo(param, mockClusterControllerClient, mockCredentials);
         assertThat(infoTest.toHiveJdbcUrl()).isEqualTo(hiveUrl);
     }
 }
```

Comment:
It would help to have test cases that check what happens if GoogleCredentials throws exceptions either in `getApplicationDefault()` and `refresh()`. E.g. `when(mockCredentials.getApplicationDefault()).thenThrow(...)`

Reply:
Testing `refresh()` makes sense, but `getApplicationDefault()` is called in `DataprocDriver`, should I create a test for it in `DataprocDriverTest`? (I think it's not necessary?)

Reply:
Yeah, a test in DataprocDriverTest would be great if there isn't one already. I think it can throw `IOException` if it can't open the credentials file (e.g. permission denied)